hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -131,80 +171,124 @@ private DifferSnapshotInfo getDSIFromSI(SnapshotInfo snapshotInfo,
         getTablePrefixes(snapshotOMMM, volumeName, bucketName));
   }
 
-  @SuppressWarnings("checkstyle:methodlength")
+  @SuppressWarnings("parameternumber")
   public SnapshotDiffReport getSnapshotDiffReport(final String volume,
                                                   final String bucket,
                                                   final OmSnapshot fromSnapshot,
                                                   final OmSnapshot toSnapshot,
                                                   final SnapshotInfo fsInfo,
-                                                  final SnapshotInfo tsInfo)
+                                                  final SnapshotInfo tsInfo,
+                                                  final int index,
```

Comment:
Related: any reason we don't want to make `token` a `long` or `int` in `message SnapshotDiffRequest`?

Reply:
It gives the flexibility to change it to encrypted or Cursor-based pagination if we need to that in future.